lib/mongo/client.rb

```diff
@@ -925,6 +927,8 @@ def database_names(filter = {}, opts = {})
     #   See https://docs.mongodb.com/manual/reference/command/listDatabases/
     #   for more information and usage.
     # @option opts [ Session ] :session The session to use.
+    # @option options [ String, Bson::Document, Hash ] :comment A user-provided
```

Comment:
Having both Hash and BSON::Document (and `BSON` needs to be capitalized there) is weird because the latter is a subclass of the former. In some of the methods, when Hash is specified what is implied is that the keys must be of the correct type (i.e. either symbols or strings - the driver will not cast Hash to B::D for indifferent access). When B::D is specified, the driver either may typecast to B::D which makes it actually accept Hash instances with wrong key types, or the driver may require actual B::D instances and fail if Hash instances are given with wrong types. So this can be confusing.

In case of comment, since the driver is not interpreting the value at all, I would tend toward documenting it as a Hash if the only two allowed types are String and Hash. If all types are allowed per Neil's comment, I suggest `Object` as the type.

Reply:
Yes, good point. Changed to `Object` since comment can be of any type.